ci-operator/step-registry/ibmcloud/provision/bastionhost/ibmcloud-provision-bastionhost-commands.sh

```diff
@@ -0,0 +1,124 @@
+workdir=${ARTIFACT_DIR}
+bastion_ignition_file="${SHARED_DIR}/${cluster_name}-bastion.ign"
+
+if [[ ! -f "${bastion_ignition_file}" ]]; then
+  echo "'${bastion_ignition_file}' not found, ignition-bastionhost step is required, abort." && exit 1
+fi
+
+VPC_CONFIG="${SHARED_DIR}/customer_vpc_subnets.yaml"
+if [[ ! -f "${VPC_CONFIG}" ]]; then
+  echo "Fail to find of VPC info file ${VPC_CONFIG}, abort." && exit 1
+fi
+echo "Reading variables from ${VPC_CONFIG}..."
+vpcName=$(yq-go r "${VPC_CONFIG}" 'platform.ibmcloud.vpcName')
+region="${LEASED_RESOURCE}"
+resource_group=$(yq-go r "${VPC_CONFIG}" 'platform.ibmcloud.resourceGroupName')
+echo "Using region: ${region}  resource_group: ${resource_group} vpc: ${vpcName}"
+
+ibmcloud_login || exit 1
+
+${IBMCLOUD_CLI} target -g ${resource_group} -r ${region}
+
+subnet=$(${IBMCLOUD_CLI} is vpc ${vpcName} --show-attached --output JSON | jq -c -r '[.subnets[] | select(.name|test("control-plane")) | .name][0]')
+
+if [ "${subnet}"X == X ]; then
+  echo "ERROR" "Fail to get subnet of vpc ${vpcName}, abort" &&  exit 1
+fi
+echo "subnet: ${subnet}"
+zone=$(${IBMCLOUD_CLI} is subnet ${subnet} --output JSON | jq -r '.zone.name')
+echo "zone ${zone}"
+
+sg=$(${IBMCLOUD_CLI} is vpc-sg ${vpcName} --output JSON | jq -r .id)
+echo "sg:" $sg
+
+#####################################
+##########Create Bastion#############
+#####################################
+run_command "${IBMCLOUD_CLI} is security-group-rule-add $sg inbound tcp --remote \"0.0.0.0/0\" --port-min=22 --port-max=22" || exit 1
+run_command "${IBMCLOUD_CLI} is security-group-rule-add $sg inbound tcp --remote \"0.0.0.0/0\" --port-min=3128 --port-max=3129" || exit 1
+run_command "${IBMCLOUD_CLI} is security-group-rule-add $sg inbound icmp --remote \"0.0.0.0/0\" --icmp-type 8 --icmp-code 0 " || exit 1
+run_command "${IBMCLOUD_CLI} is security-group-rule-add $sg inbound tcp --remote \"0.0.0.0/0\" --port-min=5000 --port-max=5000" || exit 1
+run_command "${IBMCLOUD_CLI} is security-group-rule-add $sg inbound tcp --remote \"0.0.0.0/0\" --port-min=6001 --port-max=6002" || exit 1
+
+echo "Created bastion instance"
+run_command "${IBMCLOUD_CLI} is instance-create ${bastion_name} ${vpcName} ${zone} ${MACHINE_TYPE} ${subnet} --image ${IMAGE} --user-data "@${bastion_ignition_file}" --output JSON" || exit 1
+
+sleep 60s
+
+
+#####################################
+#########Save Bastion Info###########
+#####################################
+insFile="${workdir}/${bastion_name}.json"
+run_command "${IBMCLOUD_CLI} is instance ${bastion_name} --output JSON > ${insFile}" || exit 1
+echo "INFO" "Created bastion instance ${bastion_name} status: $(jq -r '.status' ${insFile})"
+bastion_private_ip="$(jq -r '.network_interfaces[0].primary_ip.address' ${insFile})"
+nic=$(jq -r '.network_interfaces[0].id' ${insFile})
+fip="${cluster_name}-fip" 
+bastion_public_ip=$(${IBMCLOUD_CLI} is floating-ip-reserve ${fip} --nic-id $nic --output JSON | jq -r .address)
```

Comment:
How about moving the steps of fip reserving and assignment to the above section (on the top of `Save Bastion Inf`) ?

Reply:
Maybe I did not make my statement clear, suggest reorder the steps like this:
```
echo "Created bastion instance"
run_command "${IBMCLOUD_CLI} is instance-create ${bastion_name} ${vpcName} ${zone} ${MACHINE_TYPE} ${subnet} --image ${IMAGE} --user-data "@${bastion_ignition_file}" --output JSON"

sleep 300

insFile="${workdir}/${bastion_name}.json"
run_command "${IBMCLOUD_CLI} is instance ${bastion_name} --output JSON > ${insFile}"
echo "INFO" "Created bastion instance ${bastion_name} status: $(jq -r '.status' ${insFile})"

bastion_private_ip="$(jq -r '.network_interfaces[0].primary_ip.address' ${insFile})"
 
nic=$(jq -r '.network_interfaces[0].id' ${insFile})
fip="${cluster_name}-fip"
bastion_public_ip=$(${IBMCLOUD_CLI} is floating-ip-reserve ${fip} --nic-id $nic --output JSON | jq -r .address)
run_command "${IBMCLOUD_CLI} is instance-network-interface-floating-ip-add ${bastion_name} ${nic} ${fip}"

if [ X"${bastion_public_ip}" == X"" ] || [ X"${bastion_private_ip}" == X"" ] ; then
    echo "ERROR" "Failed to find bastion's public and private IP!"
    exit 1
fi

#dump the info 
run_command "${IBMCLOUD_CLI} is instance-network-interface-floating-ips ${bastion_name} $nic"
run_command "${IBMCLOUD_CLI} is sg-rules $sg --vpc ${vpcName}"

#####################################
#########Save Bastion Info###########
#####################################
echo ${bastion_private_ip} > "${SHARED_DIR}/bastion_private_address"
echo ${bastion_public_ip} > "${SHARED_DIR}/bastion_public_address"
echo "core" > "${SHARED_DIR}/bastion_ssh_user"

cat > "${bastion_info_yaml}" << EOF
publicIpAddress: ${bastion_public_ip}
privateIpAddress: ${bastion_private_ip}
bastionHost: ${bastion_name}
vpcName: ${vpcName}
EOF

proxy_credential=$(cat /var/run/vault/proxy/proxy_creds)
proxy_public_url="http://${proxy_credential}@${bastion_public_ip}:3128"
proxy_private_url="http://${proxy_credential}@${bastion_private_ip}:3128"
echo "${proxy_public_url}" > "${SHARED_DIR}/proxy_public_url"
echo "${proxy_private_url}" > "${SHARED_DIR}/proxy_private_url"
# echo proxy IP to ${SHARED_DIR}/proxyip
echo "${bastion_public_ip}" > "${SHARED_DIR}/proxyip"
```